tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/MQTTTest.java

```diff
@@ -447,6 +447,59 @@ public void testSendMoreThanUniqueId() throws Exception {
       publisher.disconnect();
    }
 
+   @Ignore
+   @Test(timeout = 600 * 1000)
+   public void testNoMessageIdReuseBeforeAcknowledgment() throws Exception {
```

Comment:
This test doesn't fail without your fix. In order to demonstrate the problem and mitigate regressions the test should fail without the fix and succeed with it.

Reply:
Fixed test by using two initial messages. Previously it would not fail as first message had ID 1 and first ID after wraparound was 2.